test/integration/multiplexed_integration_test.cc

```diff
@@ -890,6 +911,7 @@ TEST_P(Http2IntegrationTest, BadFrame) {
 // Send client headers, a GoAway and then a body and ensure the full request and
 // response are received.
 TEST_P(Http2IntegrationTest, GoAway) {
+  EXCLUDE_DOWNSTREAM_HTTP3; // QuicHttpClientConnectionImpl::goAway NOT_REACHED_GCOVR_EXCL_LINE
```

Comment:
QUICHE doesn't support client sending GOAWAY as there is no need. AFAIK Envoy doesn't support server push either, why Envoy needs this feature?

Reply:
That's fine.  I think we should test that if the API is called we don't do anything harmful (it should be a no-op like it is for HTTP/1.1, rather than crashing) and we should also test to make sure if a client sends a goaway it is a no-op or treated as a protocol error and again doesn't crash.

Reply:
ACK

Reply:
It also tests that GOAWAY does not cause existing streams to terminate.

Reply:
For HTTP/2.  For HTTP/3 given it's a protocol error to send goaway client -> server, I think it probably would cause connection termination.

Reply:
It's not a protocol error, but QUICHE client doesn't support it because QUICHE doesn't do server push. I can change NOT_REACHED_GCOVR_EXCL_LINE to log error and return. It won't affect out-going request.

Reply:
isn't it a protocol error if an H3 server gets a goaway?  cc @DavidSchinazi so I don't make things up =P

Reply:
@alyssawilk that's no longer true in IETF QUIC. Client-to-server GOAWAY tells the server to stop using server push. https://quicwg.org/base-drafts/draft-ietf-quic-http.html#name-goaway

Reply:
Ahh, thanks, I was looking at the wrong version of the spec, and Dan is of course just more up to date than me :-P